sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala

```diff
@@ -109,15 +109,11 @@ class JDBCTableCatalogSuite extends QueryTest with SharedSparkSession {
     val exp1 = intercept[NoSuchTableException] {
       sql(s"ALTER TABLE h2.test.not_existing_table RENAME TO test.dst_table")
     }
-    assert(exp1.getMessage.contains(
-      "Failed table renaming from test.not_existing_table to test.dst_table"))
-    assert(exp1.cause.get.getMessage.contains("Table \"not_existing_table\" not found"))
-    val exp2 = intercept[NoSuchNamespaceException] {
+    assert(exp1.getMessage.contains("Table test.not_existing_table not found"))
```

Comment:
This is slightly different from master: https://github.com/apache/spark/blob/37c4cd8f05316227465ff9cccbba063779827660/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala#L112-L113

since the table look up done in `DataSourceV2Strategy` instead of `Analyzer` (#30610).